homeassistant/components/unifiprotect/services.py

```diff
@@ -131,6 +151,65 @@ async def set_default_doorbell_text(hass: HomeAssistant, call: ServiceCall) -> N
     await _async_call_nvr(instances, "set_default_doorbell_message", message)
 
 
+@callback
+def _async_get_ufp_device_from_entity(
+    hass: HomeAssistant,
+    expected_type: ModelType,
+    entity: er.RegistryEntry,
+    protect: ProtectApiClient | None = None,
+) -> ProtectAdoptableDeviceModel:
+    device_id = entity.unique_id.split("_")[0]
+
```

Comment:
Does it make sense to keep a `dict` of `entity_id` -> `entry_id, ufp_device_id` in `hass.data[DOMAIN]`  that entities get added/removed from in `async_added_to_hass` and `async_will_remove_from_hass` ?